pkg/distributors/test/MultiDistributor.test.ts

```diff
@@ -885,359 +903,391 @@ describe('MultiDistributor', () => {
+                expect(previousRewardPerToken).to.be.almostEqualFp(37500);
+                const previousEarned = await distributor.getClaimableTokens(distribution, from);
 
-              await advanceTime(PERIOD_DURATION);
+                await advanceTime(PERIOD_DURATION);
 
-              // All new rewards go to user 2, meaning 45k per token
-              const currentRewardPerToken = await distributor.globalTokensPerStake(distribution);
-              expect(currentRewardPerToken).to.be.almostEqualFp(82500);
+                // All new rewards go to user 2, meaning 45k per token
+                const currentRewardPerToken = await distributor.globalTokensPerStake(distribution);
+                expect(currentRewardPerToken).to.be.almostEqualFp(82500);
 
-              const currentEarned = await distributor.getClaimableTokens(distribution, user1);
-              expect(currentEarned).to.equal(previousEarned);
-            });
+                const currentEarned = await distributor.getClaimableTokens(distribution, from);
+                expect(currentEarned).to.equal(previousEarned);
+              });
 
-            it('does not claim his rewards', async () => {
-              await distributor.unstake(stakingToken, amount, { from: user1 });
+              it('does not claim his rewards', async () => {
+                await unstake(stakingToken, amount);
 
-              // The user only got one third of the rewards for the duration they staked, which was
-              // half the period.
-              const currentRewards = await distributor.getClaimableTokens(distribution, user1);
-              expect(currentRewards).to.be.almostEqual(DISTRIBUTION_SIZE.div(2).div(3));
+                // The user only got one third of the rewards for the duration they staked, which was
+                // half the period.
+                const currentRewards = await distributor.getClaimableTokens(distribution, from);
+                expect(currentRewards).to.be.almostEqual(DISTRIBUTION_SIZE.div(2).div(3));
+              });
             });
           });
         });
+
+        context('when the user does not have the requested stake', () => {
+          const amount = fp(1001);
+
+          it('reverts', async () => {
+            await expect(unstake(stakingToken, amount)).to.be.revertedWith('UNSTAKE_AMOUNT_UNAVAILABLE');
+          });
+        });
       });
 
-      context('when the user does not have the requested stake', () => {
-        const amount = fp(1001);
+      context('when the user did not specify any amount', () => {
+        const amount = 0;
 
         it('reverts', async () => {
-          await expect(distributor.unstake(stakingToken, amount, { from: user1 })).to.be.revertedWith(
-            'UNSTAKE_AMOUNT_UNAVAILABLE'
+          await expect(unstake(stakingToken, amount)).to.be.revertedWith('UNSTAKE_AMOUNT_ZERO');
+        });
+      });
+    };
+
+    describe('unstake', () => {
```

Comment:
I'd move this before `itHnadlesUnstaking`: given how complex it is, reading how it is used before going over it seems easier. I was a bit puzzled as to what the `unstake` function meant.